pkg/querier/queryrange/roundtrip.go

```diff
@@ -89,6 +94,19 @@ func MergeMiddlewares(middleware ...Middleware) Middleware {
 	})
 }
 
+func determineCacheSplitter(cfg Config, log log.Logger) CacheSplitter {
+	if cfg.ResultsCacheConfig.SplitInterval > 0 {
+		level.Warn(log).Log("msg", "flag frontend.cache-split-interval (or config cache_split_interval) is deprecated, use querier.split-queries-by-interval instead.")
+		return cfg.ResultsCacheConfig
+	}
+	if cfg.SplitQueriesByInterval > 0 {
+		return cfg
+	}
+
+	// by default use 24 hour segments
+	return constSplitter(24 * time.Hour)
```

Comment:
This looks somewhat problematic to me because we're hiding a default setting (24h) down here instead of the CLI flag default. Moreover, removing the default value from the CLI flag has caused the default value of 24h to disappear from `config-file-reference.md`.

I still believe that trying to not introduce a breaking change is making things over complicated. What if we:
1. Completely remove `SplitInterval`
2. Add a `Validate()` to `queryrange.Config` to ensure that if `CacheResults` is `true` then `SplitQueriesByInterval` is set a well (`> 0`)
3. Keep the default `CacheResults: false` and `SplitQueriesByInterval: 0`)
I still believe that we're getting things over complicated because